src/librustdoc/html/format.rs

```diff
@@ -574,6 +574,20 @@ impl fmt::Display for clean::Impl {
     }
 }
 
+// The difference from above is that trait is not hyperlinked.
+impl clean::Impl {
+    pub fn fmt_for_trait_page(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "impl{} ", self.generics)?;
+        if let Some(ref ty) = self.trait_ {
+            write!(f, "{}{} for ",
+                   if self.polarity == Some(clean::ImplPolarity::Negative) { "!" } else { "" },
+                   ty.trait_name().unwrap())?;
+        }
+        write!(f, "{}{}", self.for_, WhereClause(&self.generics))?;
+        Ok(())
+    }
```

Comment:
Since this is just duplicating the code above, can the implementations be refactored?